examples/webserver/ratelimit/src/main/java/io/helidon/examples/webserver/ratelimit/Main.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.examples.webserver.ratelimit;
+
+import java.util.concurrent.Semaphore;
+
+import io.helidon.config.Config;
+import io.helidon.logging.common.LogConfig;
+import io.helidon.webserver.WebServer;
+import io.helidon.webserver.WebServerConfig;
+import io.helidon.webserver.http.HttpRouting;
+import io.helidon.webserver.http.ServerRequest;
+import io.helidon.webserver.http.ServerResponse;
+
+/**
+ * The application main class.
+ */
+public class Main {
+
+    private static final System.Logger LOGGER = System.getLogger(Main.class.getName());
+
+    private static Semaphore rateLimitSem = null;
```

Comment:
We should use a service instead of using static field in the main class, or a global filter if that's doable.

Reply:
Yes, I will change to use a service. Not as keen on the global filter.